optaweb-employee-rostering-frontend/src/ui/pages/rotation/SeatJigsaw.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Title, Text, Flex, FlexItem,
+  FlexModifiers, SplitItem, Split, Button,
+} from '@patternfly/react-core';
+import moment from 'moment';
+import { rosterSelectors } from 'store/roster';
+import { useSelector } from 'react-redux';
+import { EditIcon, TrashIcon } from '@patternfly/react-icons';
+import { TimeBucket } from 'domain/TimeBucket';
+import { EditTimeBucketModal } from './EditTimeBucketModal';
+import { Stub, EmployeeNickName } from './EmployeeStub';
+
+/* tslint:disable:no-nested-ternary react/no-array-index-key */
```

Comment:
Seems to be unused.

Reply:
Nested Ternary here:
` backgroundColor: seat ? seat.employee ? seat.employee.color : 'white' : 'gray',`

No array index key used here:
            {new Array(Math.ceil(rotationLength / 7)).fill(null).map((_, index) => index).map(weekNumber => (
              <FlexItem key={`${weekNumber}`}>

Reply:
- ternary - already disabled with `// eslint-disable-next-line no-nested-ternary` on line 84. No need to disable it for the whole file.
- index key - the example on line 68 doesn't use the index as the key. `weekNumber` is not the index, it's the element. So the "no array index key" rule is not broken in this case.

Also note that this line (30) uses a _tslint_ ignore comment but TSLint is obsolete. We no longer use TSLint.